workflow/controller/operator.go

```diff
@@ -807,6 +812,16 @@ func (woc *wfOperationCtx) checkReconciliationComplete() bool {
 	return woc.wf.Status.Phase.Completed() && !woc.wf.Status.TaskResultsInProgress()
 }
 
+func (woc *wfOperationCtx) checkWorkflowComplete() bool {
+	woc.log.Debugf("Task results completion status: %v", woc.wf.Status.TaskResultsCompletionStatus)
+	return woc.wf.Labels[common.LabelKeyCompleted] == "true" && !woc.wf.Status.TaskResultsInProgress()
+}
+
+func (woc *wfOperationCtx) checkTaskResultsComplete() bool {
+	woc.log.Debugf("Task results completion status: %v", woc.wf.Status.TaskResultsCompletionStatus)
+	return !woc.wf.Status.TaskResultsInProgress()
+}
```

Comment:
we can't call this `checkTaskResultsComplete()` since this can be true when some TaskResults haven't started

Reply:
Ok, I have changed it.Thanks

Reply:
Why did you end up putting this back? (per previous comment)